panoptes_client/workflow.py

```diff
@@ -530,6 +532,73 @@ def configure_for_alice(self):
         self.add_alice_reducers()
         self.add_alice_rules_and_effects()
 
+    def run_aggregation(self, user=None, delete_if_exists=False):
+        """
+        This method will start a new batch aggregation run, Will return a dict with the created aggregation if successful.
+
+        - **user** can be either a :py:class:`.User` or an ID.
+        - **delete_if_exists** parameter is optional if true, deletes any previous instance
+        -
+        Examples::
+
+            Workflow(1234).run_aggregation(1234)
+            Workflow(1234).run_aggregation(user=1234, delete_if_exists=True)
+        """
+
+        if(isinstance(user, User)):
+            _user_id = user.id
+        elif (isinstance(user, (int, str,))):
+            _user_id = user
+        else:
+            raise TypeError('Invalid user parameter')
+
+        try:
+            workflow_aggs = self.get_batch_aggregations()
+            if workflow_aggs.object_count > 0:
+                agg_id = workflow_aggs.next().id
+                current_wf_agg = Aggregation.find(agg_id)
+                if delete_if_exists:
+                    current_wf_agg.delete()
+                    return self._create_agg(_user_id)
+                else:
+                    return current_wf_agg
+            else:
+                return self._create_agg(_user_id)
+        except PanoptesAPIException as err:
+            raise err
+
+    def get_batch_aggregations(self):
+        return Aggregation.where(workflow_id=self.id)
+
+    def _create_agg(self, user_id):
+        new_agg = Aggregation()
+        new_agg.links.workflow = self.id
+        new_agg.links.user = user_id
+        new_agg.save()
+        return new_agg
+
+    def _get_agg_property(self, param):
+        try:
+            aggs = self.get_batch_aggregations()
+            next = six.next(aggs)
```

Comment:
Possibly stylistic, but I'm not the biggest fan of declaring a variable `next`, mainly because i worry it rewrites Python's built-in `next` method. We could rename this variable to `agg` or something. 

Though, I think we could get away with 
`return getattr(next(aggs), param, None)` or something like that